packages/app/src/components/Bookmarks/BookmarkFolderItem.tsx

```diff
@@ -162,13 +167,87 @@ const BookmarkFolderItem: FC<BookmarkFolderItemProps> = (props: BookmarkFolderIt
     mutateBookmarkInfo();
   }, [mutateBookmarkInfo, mutateParentBookmarkFolder]);
 
+  const [, bookmarkFolderDragRef] = useDrag({
+    type: 'FOLDER',
+    item: props,
+    end: (item, monitor) => {
+      const dropResult = monitor.getDropResult();
+      if (dropResult != null) {
+        mutateParentBookmarkFolder();
+      }
+    },
+    collect: monitor => ({
+      isDragging: monitor.isDragging(),
+      canDrag: monitor.canDrag(),
+    }),
+  });
+
+
+  const folderItemDropHandler = async(item: BookmarkFolderItemProps) => {
+    try {
+      await apiv3Put('/bookmark-folder', { bookmarkFolderId: item.bookmarkFolder._id, name: item.bookmarkFolder.name, parent: bookmarkFolder._id });
+      await mutateChildBookmarkData();
+      toastSuccess(t('toaster.update_successed', { target: t('bookmark_folder.bookmark_folder') }));
+    }
+    catch (err) {
+      toastError(err);
+    }
+  };
+
+  const bookmarkItemDropHandler = useCallback(async(item: IPageHasId) => {
+    try {
+      await apiv3Post('/bookmark-folder/add-boookmark-to-folder', { pageId: item._id, folderId: bookmarkFolder._id });
+      mutateParentBookmarkFolder();
+      toastSuccess('Bookmark added to bookmark folder successfully');
+    }
+    catch (err) {
+      toastError(err);
+    }
+
+  }, [bookmarkFolder._id, mutateParentBookmarkFolder]);
+
+
+  const isDroppable = (item: BookmarkFolderItemProps, targetRoot: string, targetLevel: number): boolean => {
+    if (item.bookmarkFolder.parent === bookmarkFolder._id || item.bookmarkFolder._id === bookmarkFolder._id) {
+      return false;
+    }
+    if (item.root === targetRoot) {
+      if (item.level < targetLevel) {
+        return false;
+      }
+    }
+    return true;
```

Comment:
```
if (item.root === targetRoot) {
  if (item.level < targetLevel) {
    return false;
  }
}
return true;
```

It is possible to write this way, if you like

`return item.root !== targetRoot || item.level >= targetLevel;`

Reply:
I just implemented this line change